runtime/compiler/control/CompilationThread.cpp

```diff
@@ -7509,7 +7509,8 @@ TR::CompilationInfoPerThreadBase::preCompilationTasks(J9VMThread * vmThread,
              !entry->isJNINative() &&
              !details.isNewInstanceThunk() &&
              !details.isMethodHandleThunk() &&
-             TR::Options::getCmdLineOptions()->getOption(TR_EnableJITServerHeuristics) &&
+             (TR::Options::getCmdLineOptions()->getOption(TR_EnableJITServerHeuristics) ||
```

Comment:
Let's another check that the optimization level is greater than cold, to avoid cases where we change from cold to cold. In the past this was not needed because under `enableJITHeuristics` all cold compilations were already local, so if we entered this 'if' statement it was guaranteed that we had a warm/hot/scorching compilation. Currently, this is no longer true.

I don't remember why we needed that test for `!details.isMethodHandleThunk()`. I know that `!entry->isJNINative()` `!details.isNewInstanceThunk()` are needed because those are not recompilable.

Reply:
Base on this code
https://github.com/eclipse-openj9/openj9/blob/0831118bc92e59571099e14b2928e3e50c4c19e4/runtime/compiler/control/CompilationThread.cpp#L8859-L8865,

method handle thunks are not recompilable as well.